Framework/Algorithms/test/PolarizationCorrectionWildesTest.h

```diff
@@ -1475,6 +1378,152 @@ class PolarizationCorrectionWildesTest : public CxxTest::TestSuite {
+
+    const auto effWS = efficiencies(edges);
+    auto alg = createWildesAlg(wsNames, effWS, grouping);
+    alg->setProperty("AddSpinStateToLog", addSpinStateToLog);
+
+    const auto outputWS = runAlg(std::move(alg));
+    validateNumberOfEntries(outputWS, numClones);
+
+    if (addSpinStateToLog) {
+      validateSpinStateLogs(outputWS, expectedLogValues);
+    } else {
+      validateNoSpinStateLogs(outputWS, numClones);
+    }
+  }
+
+  MatrixWorkspace_sptr createWorkspace(size_t nHist = 2, std::optional<BinEdges> edges = std::nullopt,
+                                       std::optional<Counts> counts = std::nullopt) {
+    if (!edges) {
+      edges = BinEdges{0.3, 0.6, 0.9, 1.2};
+    }
+
+    const auto yVal = 2.3;
+    if (!counts) {
+      counts = Counts{yVal, 4.2 * (yVal), yVal};
+    }
+
+    return create<Workspace2D>(nHist, Histogram(*edges, *counts));
+  }
+
+  MatrixWorkspace_sptr createWorkspace(size_t nHist, const Counts &counts) {
+    return createWorkspace(nHist, std::nullopt, counts);
+  }
+
+  MatrixWorkspace_sptr createWorkspace(const Counts &counts) { return createWorkspace(2, std::nullopt, counts); }
+
+  std::vector<MatrixWorkspace_sptr> cloneWorkspaces(const MatrixWorkspace_sptr &sourceWorkspace,
+                                                    const size_t numClones) {
+    std::vector<MatrixWorkspace_sptr> clonedWorkspaces;
+
+    clonedWorkspaces.push_back(sourceWorkspace);
+    for (size_t i = 1; i < numClones; ++i) {
+      clonedWorkspaces.push_back(sourceWorkspace->clone());
+    }
+
+    return clonedWorkspaces;
+  }
+
+  std::vector<std::string> generateWorkspaceNames(const size_t numClones) {
+    std::vector<std::string> wsNames;
+
+    for (size_t i = 0; i < numClones; ++i) {
+      wsNames.push_back("ws" + std::to_string(i));
+    }
+
+    return wsNames;
+  }
+  void addWorkspacesToService(const std::vector<std::string> &wsNames, const std::vector<MatrixWorkspace_sptr> &wsList,
+                              const size_t numWorkspaces) {
+    for (size_t i = 0; i < numWorkspaces; ++i) {
+      AnalysisDataService::Instance().addOrReplace(wsNames[i], wsList[i]);
+    }
+  }
+
+  void addWorkspaceToService(const std::string &wsName, const MatrixWorkspace_sptr &ws) {
+    AnalysisDataService::Instance().addOrReplace(wsName, ws);
+  }
+
+  void prepareAndRegisterWorkspaces(std::vector<std::string> &wsNames,
+                                    const std::vector<Mantid::API::MatrixWorkspace_sptr> &wsList, const size_t nHist) {
+
+    for (size_t i = 0; i != wsNames.size(); ++i) {
+      for (size_t j = 0; j != nHist; ++j) {
+        wsList[i]->mutableY(j) *= static_cast<double>(i + 1);
+        wsList[i]->mutableE(j) *= static_cast<double>(i + 1);
+      }
+      addWorkspacesToService(wsNames, wsList, wsNames.size());
+    }
+  }
+
+  std::vector<MatrixWorkspace_sptr> createWorkspaceList(const std::optional<Counts> counts, const size_t numClones) {
```

Comment:
```suggestion
  std::vector<MatrixWorkspace_sptr> createWorkspaceList(size_t numClones, const std::optional<Counts>& counts = std::nullopt ) {
```
Using this function signature allows you to call ```createWorkspaceList``` from the test using only the ```numClones``` parameter, and avoid using the other ```createWorkspaceList``` function.